src/lib/api.js

```diff
@@ -229,15 +229,25 @@ export default class RocketChatInstance {
   }
 
   async realtime(callback) {
+    let subscriptionId;
     try {
       await this.rcClient.connect();
       await this.rcClient.resume({ token: Cookies.get(RC_USER_TOKEN_COOKIE) });
-      await this.rcClient.subscribe('stream-room-messages', this.rid);
-      await this.rcClient.onMessage((data) => {
+      subscriptionId = await this.rcClient.subscribe(
+        'stream-room-messages',
+        this.rid
+      );
+
+      const messageHandler = (data) => {
+        this.rcClient.unsubscribe(subscriptionId);
         callback(data);
-      });
+      };
+
+      await this.rcClient.onMessage(messageHandler);
```

Comment:
is `onMessage` room specific or called everytime?

Reply:
it is called for all messages received by the client and  The streamDataHandler function is also not room-specific, but it filters events based on the room ID before calling the callback.